### PR TITLE
dynamic URIs, better input examples

### LINKED ISSUE #7: Pass in URL options via record instead of config
Example URL

'ses://secret@autoidm.com/secret/secret/us-east-2/secret@autoidm.org/?name="Secret Notifications"&cc="secret@secret.org"&bcc="secret@secret.com"'

You may want to be able to pass in arguments like Name, CC, and BCC via the record instead of a configuration. This could be accomplish with a key called "Options" in the record, it'd be a JSON object where we pass these options along and append them to the URL.

### PR DESCRIPTION
**Pull request by taliasmiley**
Closes #7

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,9 +11,9 @@ license = "Apache 2.0"
 readme = "README.md"
 
 [tool.poetry.dependencies]
-python = "<3.11,>=3.7.1"
+python = "<3.12,>=3.8"
```
Comment (sebastianswms) [inline pyproject.toml:14]: breaking change

Comment (visch) [reply pyproject.toml:14]: I wouldn't hard bump the version number for this but 🤷

File: target_apprise/sinks.py [hunk 2]
```diff
@@ -8,15 +9,48 @@
...
+
+                pattern = r"\{_sdc_replace_.*?\}"
+                if re.search(pattern, formatted_uri):
+                    debug_msg = (  # Contains secrets!
+                        "Encountered an error while attempting to do URI "
+                        f"replacement for URI: '{uri}'."
+                    )
+                    self.logger.debug(debug_msg)
+                    msg = (
+                        "At least one of the configured `uris` contains "
```
Comment (visch) [inline target_apprise/sinks.py:45]: ? this isn't used right?

**Review by visch — CHANGES_REQUESTED** [review]

**Comment by visch** [comment]
Everything else looks good

File: pyproject.toml (hunk 1 above)
Comment (sebastianswms) [reply pyproject.toml:14]: Newer versions of the SDK no longer support 3.7.

**Pull request merged**